### PR TITLE
Fix/convert model

### PR DESCRIPTION
**Pull request by wenshuoliu**
Changes based on our discussion:
- Add skip connection and layer norm as options to DenseStack; activation can be a list. For both pytorch and tf. 
- Move `create_model` to `transformer_utils.py`. Keep the name since it also creates seq2seq model.
- Put `SingleHeadReduction` in eight mile layers, considering it's a fancy pooling and might be useful somewhere else. 
- `response-selection.py` only works for dual-encoder model right now for simplicity. 

Changes slightly different from our discussion:
- Keep `TwoHeadConcat` in `transformer_utils.py` since it's very specific to dual-encoder model. 
- Add a customized module `ConveRTFFN` to still implement the convert graph. Looking at it again, I think the convert graph does make some sense, though not compatible with the DenseStack we want. E.g. the reason of layer norms at both beginning and end is because the `final` layer doesn't have either activation or layer norm. And the "projecting skip-connection" adds two unnormalized tensor before going into the final layer norm. I don't want to make much changes to the model structure since I've been experimenting on it.

### CODE REVIEW

File: api-examples/transformer_utils.py [hunk 1]
```diff
@@ -407,3 +458,29 @@ def forward(self, inputs, targets):
                 pred = self.model(in_)
                 return self._loss(pred, targets)
         return LossFn(self, loss)
+
+
+def create_model(embeddings, d_model, d_ff, dropout, num_heads, num_layers, model_type, rpr_k, d_k, reduction_d_k,
```
Comment (dpressel) [inline api-examples/transformer_utils.py:463]: create_paired_model?

File: baseline/pytorch/embeddings.py [hunk 2]
```diff
@@ -208,11 +208,15 @@ def _identity(x):
     return x
 
 
-def _mean_pool(_, embeddings):
+def _mean_pool(inputs, embeddings):
```
Comment (dpressel) [inline baseline/pytorch/embeddings.py:211]: tf impl?

File: baseline/pytorch/embeddings.py [hunk 3]
```diff
@@ -208,11 +208,15 @@ def _identity(x):
...
+    mask = (inputs != 0)
+    embeddings = embeddings * mask.unsqueeze(-1)
     return torch.mean(embeddings, 1, False)
 
 
-def _max_pool(_, embeddings):
+def _max_pool(inputs, embeddings):
```
Comment (dpressel) [inline baseline/pytorch/embeddings.py:217]: dont really need this I think

File: baseline/pytorch/embeddings.py [hunk 4]
```diff
@@ -233,8 +237,10 @@ def __init__(self, name, **kwargs):
             self.pooling_op = self._cls_pool
 
     def _sqrt_length_pool(self, inputs, embeddings):
```
Comment (dpressel) [inline baseline/pytorch/embeddings.py:239]: tf?

File: api-examples/transformer_utils.py (hunk 1 above)
Comment (wenshuoliu) [reply api-examples/transformer_utils.py:463]: this also create seq2seq model, so i'd like to keep it

Comment (dpressel) [reply api-examples/transformer_utils.py:463]: seq2seq is still paired.  the problem is the current name is not specific

**Comment by dpressel** [comment]
Im going to close this for now to cut another RC, but please remove the sqrt pool from the pytorch model and rename `create_model` to `create_paired_model` when you get a chance.

**Pull request merged**